icu4c/source/i18n/persncal.cpp

```diff
@@ -109,9 +109,10 @@ int32_t PersianCalendar::handleGetLimit(UCalendarDateFields field, ELimitType li
  */
 UBool PersianCalendar::isLeapYear(int32_t year)
 {
-    int32_t remainder;
-    ClockMath::floorDivide(25 * year + 11, 33, remainder);
-    return (remainder < 8);
+    int32_t cte = 683 / 2820;
```

Comment:
Apart from the problems with the algorithm, this code doesn't do what it tries to do. For example, `cte` is set to zero here, since it's an `int32_t`.

Reply:
Thanks for mentioning this problem. I don't know C++ well, so I guess changing it to `float32_t` might fix this problem. Please let me know if I'm right.

Reply:
This would still not work, since the division of 683 over 2820 is done as integers. You need to change 683 to 683.0 or something like that. Also there's no float32_t in C.